apps/prairielearn/src/pages/instructorAssessmentAccessOverrides/instructorAssessmentAccessOverrides.js

```diff
@@ -0,0 +1,137 @@
+// @ts-check
+import * as express from 'express';
+const asyncHandler = require('express-async-handler');
+import * as sqldb from '@prairielearn/postgres';
+import * as error from '@prairielearn/error';
+import { getEnrollmentForUserInCourseInstance } from '../../models/enrollment';
+import { selectUserByUid } from '../../models/user';
+
+const router = express.Router();
+const sql = sqldb.loadSqlEquiv(__filename);
+
+async function getUserOrGroupId({ course_instance_id, assessment, uid, group_name }) {
+  if (assessment.group_work) {
+    if (!group_name || uid) {
+      throw error.make(400, 'Group name is required for group work assessments.');
+    }
+    const group_result = await sqldb.queryZeroOrOneRowAsync(sql.select_group_in_assessment, {
+      group_name: group_name,
+      course_instance_id: course_instance_id,
+      assessment_id: assessment.id,
+    });
+    if (group_result.rows.length > 0) {
+      return { user_id: null, group_id: group_result.rows[0].id };
+    } else {
+      throw error.make(400, 'Group not found in this assessment.');
+    }
+  }
+  if (uid) {
+    if (group_name) {
+      throw error.make(400, 'Student UID is required for individual work assessments.');
+    }
+    const user = await selectUserEnrolledInCourseInstance({
+      uid: uid,
+      course_instance_id: course_instance_id,
+    });
+    if (!user) {
+      throw error.make(400, `User ${uid} is not enrolled in this course instance.`);
+    }
+
+    return { user_id: user.user_id, group_id: null };
+  } else {
+    throw error.make(400, 'Student UID or Group Name is required.');
+  }
+}
+
+async function selectUserEnrolledInCourseInstance({ uid, course_instance_id }) {
+  const user = await selectUserByUid(uid);
+  if (!user) return null;
+
+  const enrollment = await getEnrollmentForUserInCourseInstance({
+    user_id: user.user_id,
+    course_instance_id: course_instance_id,
+  });
+  if (!enrollment) return null;
+
+  return user;
+}
+
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_instance_permission_view) {
+      throw error.make(403, 'Access denied (must be a student data viewer)');
+    }
+    const result = await sqldb.queryAsync(sql.select_assessment_access_policies, {
+      assessment_id: res.locals.assessment.id,
+    });
+    res.locals.policies = result.rows;
+    res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+  }),
+);
+
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_instance_permission_view) {
```

Comment:
You had this conditional correct in a previous commit but then changed it. Remember this is for a `POST` handler that will allow folks to make changes. We don't want someone with only _view_ permissions to be able to edit, they must have _edit_ permissions to do so.

Reply:
Done in [a36f2d2](https://github.com/PrairieLearn/PrairieLearn/pull/9103/commits/a36f2d2018227ffe5dee693855ae0ca7fb2f516b)

Reply:
Tabs converted in [9ef2171](https://github.com/PrairieLearn/PrairieLearn/pull/9103/commits/9ef2171585c3daafdb5f70f9221daa19cf8a085a)